GitHubRepositories.ps1

```diff
@@ -191,6 +191,125 @@ function New-GitHubRepository
     return Invoke-GHRestMethod @params
 }
 
+function New-GitHubRepositoryFromTemplate
+{
+<#
+    .SYNOPSIS
+        Creates a new repository on GitHub from a template repository.
+
+    .DESCRIPTION
+        Creates a new repository on GitHub from a template repository.
+
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER RepositoryName
+        Name of the repository to be created.
+
+    .PARAMETER TemplateOwnerName
+        Owner of the template repository.
+
+    .PARAMETER TemplateRepositoryName
+        Name of the template repository.
+
+    .PARAMETER OwnerName
+        Owner of the repository to be created. If not specified, the DefaultOwnerName configuration
+        property value will be used.
+
+    .PARAMETER Description
+        A short description of the repository.
+
+    .PARAMETER Private
+        By default, this repository will created Public.  Specify this to create a private
+        repository.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        New-GitHubRepositoryFromTemplate -RepositoryName MyNewRepo -TemplateOwnerName MyOrg -TemplateRepositoryName MyTemplateRepo -OwnerName Me
+
+        Creates a new GitHub repository from the specified template repository
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        PositionalBinding = $false
+        )]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "",
```

Comment:
I know that this hasn't bee updated post-pipeline work yet.  As a reminder though, please add:
  * `.INPUTS`
  * `.OUTPUTS`
  * `[OutputType()]

and update the function to be a `filter`.